### PR TITLE
Vehicle main base

### PR DESCRIPTION
**Pull request by stuckatmarine**
- Base srauv_main.py created to iterate on
- Msg pipeline in place with intermediate websocket server  
- Minor sim updates?
- Multiprocessing/threading/socketing Examples/ dir
- Custom logging and sample logs

### CODE REVIEW

File: pi4_software/srauv_main.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+dist_sensor_values = [0.0, 0.0, 0.0, 0.0, 0.0]
+ds_threads = []
+thrust_values = [0, 0, 0, 0, 0, 0] # int, -100 to 100 as percent max thrust
+th_threads = []
+internal_socket_threads = []
+
+cmd_msg = command_msg.make(source, "sim")
+tel_msg = telemetry_msg.make(source, "sim")
+
+log_filename = str(f'Logs/{datetime.now().strftime("SR--%m-%d-%Y_%H-%M-%S")}.log')
```
Comment (andrewnash) [inline pi4_software/srauv_main.py:56]: Might make sense to let the logger handle the creation of the log file name and just pass in "SR", "IS", etc

File: pi4_software/srauv_main.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+
+def apply_thrust(enable):
+    logger.info(f"thrust_enabled:{th_threads[0].thrust_enabled} thrust_values:{thrust_values}")
+    for t in th_threads:
+        t.do_thrust(enable)
+
+
+def has_live_threads(threads):
+    return True in [t.is_alive() for t in threads]
```
Comment (andrewnash) [inline pi4_software/srauv_main.py:142]: beautiful line

**Review by andrewnash — COMMENTED** [review]
Minor comments

File: pi4_software/Examples/middleware_server.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        while not self.kill_received:
+            try:
+                data, address = self.sock.recvfrom(4096)
+                
+                if data:
+                    logger.info(f"addr:{address} data:{data}")
+                    json_data = {"source": ''}
+
+                    if not self.inbound_queue_dict():
+                        json_data = json.loads(self.inbound_queue.get().decode('utf8').replace("'", '"'))
```
[the remark this answers is not in the document]
Comment (stuckatmarine) [reply pi4_software/Examples/middleware_server.py:51]: de-queues the front value, but not plannning on using this middleware example anymore

File: pi4_software/external_ws_server.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    def handleMessage(self):
+        try:
+            # Received ws msg from external network. Log it
+            logger.info(f"sending to {srauv_address}  data:{self.data}")
+
+            # Forward msg to srauv_main's local socket
+            srauv_socket.sendto(self.data.encode("utf-8"), srauv_address)
+
+            # Get response from srauv_main. Log it
+            data, addr = srauv_socket.recvfrom(4096)
```
[the remark this answers is not in the document]
Comment (stuckatmarine) [reply pi4_software/external_ws_server.py:44]: they shouldnt have images for the vehicle server, on the PI, if we rely on gstreamer to pass the video around. Image stuff was just added for your testing.

Can increase this if need though

**Pull request merged**